spec/view_component/form/check_box_component_spec.rb

```diff
@@ -0,0 +1,22 @@
+# frozen_string_literal: true
+
+RSpec.describe ViewComponent::Form::CheckBoxComponent, type: :component do
+  let(:object)  { OpenStruct.new }
+  let(:form)    { form_with(object) }
+  let(:options) { {} }
+
+  let(:component) { render_inline(described_class.new(form, object_name, :admin, "1", "0", options)) }
+  let(:component_html_attributes) { component.css("input").last.attributes }
+
+  context "with simple args" do
+    it do
+      expect(component.to_html).to eq(
+        %(<input name="user[admin]" type="hidden" value="0">) +
+          %(<input type="checkbox" value="1" name="user[admin]" id="user_admin">)
```

Comment:
@nicolas-brousse idea for a better syntax? (without newlines and whitespace between the `input`s)

Reply:
In our current project, I use to have this kind of tests.

```rb
      expect(component.to_html.tr("\n", "")).to eq <<~HTML.tr("\n", "").strip
        <label class="form-checkbox" for="user_is_enabled">
        <input name="user[is_enabled]" type="hidden" value="0">
        <input class="form-checkbox__input" type="checkbox" value="1" name="user[is_enabled]" id="user_is_enabled">
        <span class="form-checkbox__label">Actif</span>
        </label>
      HTML
```

It's not the cleanest. Maybe we may create a custom matcher like
```rb
      expect(component).to have_html <<~HTML
        <label class="form-checkbox" for="user_is_enabled">
        <input name="user[is_enabled]" type="hidden" value="0">
        <input class="form-checkbox__input" type="checkbox" value="1" name="user[is_enabled]" id="user_is_enabled">
        <span class="form-checkbox__label">Actif</span>
        </label>
      HTML
```

Reply:
Yes, let's create a custom matcher, can you open a PR for it?

Reply:
Sure.